app/model/Country.scala

```diff
@@ -0,0 +1,268 @@
+  case FR extends Country("FR", "France")
+  case CL extends Country("CL", "Chile")
+  case ID extends Country("ID", "Indonesia")
+  case TJ extends Country("TJ", "Tajikistan")
+  case AG extends Country("AG", "Antigua & Barbuda")
+  case TD extends Country("TD", "Chad")
+  case ME extends Country("ME", "Montenegro")
+  case PG extends Country("PG", "Papua New Guinea")
+  case KN extends Country("KN", "Saint Christopher & Nevis")
+  case IM extends Country("IM", "Isle of Man")
+  case TV extends Country("TV", "Tuvalu")
+  case ET extends Country("ET", "Ethiopia")
+  case SD extends Country("SD", "Sudan")
+  case NR extends Country("NR", "Nauru")
+  case TO extends Country("TO", "Tonga")
+  case NL extends Country("NL", "Netherlands")
+  case UG extends Country("UG", "Uganda")
+  case TF extends Country("TF", "French Southern Territories")
+  case MR extends Country("MR", "Mauritania")
+  case DO extends Country("DO", "Dominican Republic")
+  case KG extends Country("KG", "Kyrgyzstan")
+  case MF extends Country("MF", "Saint Martin")
+  case MH extends Country("MH", "Marshall Islands")
+  case SB extends Country("SB", "Solomon Islands")
+  case MY extends Country("MY", "Malaysia")
+  case MZ extends Country("MZ", "Mozambique")
+  case TR extends Country("TR", "Turkey")
+  case GU extends Country("GU", "Guam")
+  case TZ extends Country("TZ", "Tanzania")
+  case CH extends Country("CH", "Switzerland")
+  case AE extends Country("AE", "United Arab Emirates")
+  case GS extends Country("GS", "South Georgia & The South Sandwich Islands")
+  case PS extends Country("PS", "Palestinian Territories")
+  case KM extends Country("KM", "Comoros")
+  case BF extends Country("BF", "Burkina Faso")
+  case GY extends Country("GY", "Guyana")
+  case CF extends Country("CF", "Central African Republic")
+  case LS extends Country("LS", "Lesotho")
+  case CY extends Country("CY", "Cyprus")
+  case AW extends Country("AW", "Aruba")
+  case NP extends Country("NP", "Nepal")
+  case DE extends Country("DE", "Germany")
+  case AQ extends Country("AQ", "Antarctica")
+  case AM extends Country("AM", "Armenia")
+  case FJ extends Country("FJ", "Fiji")
+  case LC extends Country("LC", "Saint Lucia")
+  case CI extends Country("CI", "Ivory Coast")
+  case TM extends Country("TM", "Turkmenistan")
+  case KY extends Country("KY", "Cayman Islands")
+  case BD extends Country("BD", "Bangladesh")
+  case CO extends Country("CO", "Colombia")
+  case PF extends Country("PF", "French Polynesia")
+  case KP extends Country("KP", "North Korea")
+  case BS extends Country("BS", "Bahamas")
+  case PT extends Country("PT", "Portugal")
+  case GN extends Country("GN", "Guinea")
+  case CD extends Country("CD", "Congo (Kinshasa)")
+  case HU extends Country("HU", "Hungary")
+}
+
+object Country {
+  private val countriesByName: Map[String, Country] = Country.values.map(c => c.name -> c).toMap
+
+  def getCountryByName(name: String): Option[Country] = countriesByName.get(name)
```

Comment:
I think `find` is the more idiomatic way to look up something that might not be there:
```suggestion
  def findByName(name: String): Option[Country] = countriesByName.get(name)
```

Reply:
done